### PR TITLE
[Concept Entry] Rust Data Types #3373

### PR DESCRIPTION
**Pull request by moha230**
### Description

- added concept entry rust data-types
- added a folder data-types 
- added a file data-types.md
- Closes issue  #3373 

### Type of Change

- Adding concept entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR

### CODE REVIEW

File: content/rust/concepts/data-types/data-types.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+Title: 'Data-Types'
+Description: 'A brief description.'
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:3]: Make this desc a small introduction or definition what data types are

File: content/rust/concepts/data-types/data-types.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+**Data types** serve to define the nature of data a variable can store. Rust employs static typing, ensuring that variable types are known at compile time, aiding in the detection of potential errors before runtime. These data types are broadly categorized into two groups: primitive types and compound types.
+
+## Primitive types || scala types
+
+Scala types represent a single value.
+
+- Integer
+- Floating-Point
+- Boolean
+- Character
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:24]: give small definition what these are please

File: content/rust/concepts/data-types/data-types.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Compound Types
+
+compund types represent multiple values and can group them together.
+
+- tuples
+- arrays
```
Comment (dakshdeepHERE) [inline content/rust/concepts/data-types/data-types.md:31]: do same for these too smal desc will be good. Also check the rust entries see if there are tuples and arrays entry link them here.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Few changes for now work on them first then we will do the reviews of codebytes @moha230

File: content/rust/concepts/data-types/data-types.md (hunk 1 above)
Comment (moha230) [reply content/rust/concepts/data-types/data-types.md:3]: Sorry i had not seen that

Comment (dakshdeepHERE) [reply content/rust/concepts/data-types/data-types.md:3]: @moha230 you still havent fixed the description

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I am still waiting for you to work on the suggested changes @moha230. Please work on the commits so we can move forward with the reviews

**Comment by moha230** [comment]
Hello @dakshdeepHERE its done Thank you

**Review by dakshdeepHERE — APPROVED** [review]
LGTM 🚀

**Comment by ishg-153** [comment]
Hey @moha230, sorry for the delay. You seem to have merged `Codecademy:main` into your branch which seems to have added files that are unrelated to the PR and its corresponding issue.

**Comment by moha230** [comment]
Hello I noticed there was a problem but I did not know what was the mistake . Is there a way of fixing it's please advise thank you @ishg-153

**Comment by ishg-153** [comment]
Yes @moha230, you will need to open a new PR.

I'd suggest you use `cherry-pick`.

Make sure to create a new branch from the main branch and then `cherry-pick` only the commits that you made to add or make changes to the `content/rust/concepts/data-types/data-types.md` file. Before creating the new branch make sure that your remote and local repos are up-to-date and you have a clean working tree to avoid any issues.

After that create a new PR from that branch. Let me know once you are done so that I can close down this PR.

Now if you want to clean your working tree you can go ahead and delete `data-types` branch (only do this if cherry-pick works and do it after you create a new PR from the new branch).

**Comment by ishg-153** [comment]
Let me know if you need more help.

**Comment by ishg-153** [comment]
@dakshdeepHERE should we close this issue?

**Comment by dakshdeepHERE** [comment]
Yes @ishg-153 lets close this PR also the new one isn't right. I have given the author a simple fix to follow. I think creating a totally new branch off the main branch would help him.

**Pull request closed without merging**